lib/Serge/Engine/Plugin/parse_yaml.pm

```diff
@@ -134,12 +141,30 @@ sub parse {
         die $error_text;
     }
 
+    if ($self->{data}->{yaml_mode} eq 'rails') {
+        my @tree_keys = keys(%$tree);
+        my $tree_count = int(@tree_keys);
+        if ($tree_count == 0) {
+            # Special case NOOP. Empty data
+        } elsif ($tree_count == 1) {
+            $tree = $tree->{$tree_keys[0]};
+        } else {
+            my $err = join(', ', @tree_keys);
+            $err = "YAML file with more than 1 key at root in yaml_mode: rails: $err";
+            $self->{errors}->{$self->{parent}->{engine}->{current_file_rel}} = $err;
+            # Or should we die?
+            warn "WARNING: $err";
+        }
+    }
+
     # Process tree recursively
 
     $self->process_node('', $tree, $callbackref, $lang);
 
     # Reconstruct YAML
 
+    $tree = {$lang => $tree} if ($lang && $self->{data}->{yaml_mode} eq 'rails');
```

Comment:
> I think that the default region/country per language should probably be configurable, since different users will want to map those differently. (Notably ar => ar-AE, but also hi => hi-IN is missing). I will file a separate issue for that.]

Users can simply use `ar-ae` in the list of destination languages, or use [output_lang_rewrite](https://serge.io/docs/configuration-files/reference/#sel:I2NvZGVfMC9zcGFuLjE5NDI4L0AtI2NvZGVfMC9zcGFuLjE5NDQ2L0A6MQ) parameter. So this is already pretty configurable.